src/types/onyx/Policy.ts

```diff
@@ -68,18 +69,101 @@ type TaxRatesWithDefault = {
     taxes: TaxRates;
 };
 
-// These types are for the Integration connections for a policy (eg. Quickbooks, Xero, etc).
-// This data is not yet used in the codebase which is why it is given a very generic type, but the data is being put into Onyx for future use.
-// Once the data is being used, these types should be defined appropriately.
-type ConnectionLastSync = Record<string, unknown>;
-type ConnectionData = Record<string, unknown>;
-type ConnectionConfig = Record<string, unknown>;
-type Connection = {
+type ConnectionLastSync = {
+    successfulDate?: string;
+    errorDate?: string;
+    isSuccessful: boolean;
+    source: 'DIRECT' | 'EXPENSIFYWEB' | 'EXPENSIFYAPI' | 'AUTOSYNC' | 'AUTOAPPROVE';
+};
+
+type Account = {
+    glCode?: string;
+    name: string;
+    currency: string;
+    id: string;
+};
+
+type Employee = {
+    id: string;
+    firstName?: string;
+    lastName?: string;
+    name: string;
+    email: string;
+};
+
+type Vendor = {
+    id: string;
+    name: string;
+    currency: string;
+    email: string;
+};
+
+type TaxCode = {
+    totalTaxRateVal: string;
+    simpleName: string;
+    taxCodeRef: string;
+    taxRateRefs: Record<string, string>;
+    name: string;
+};
+
+type QBOConnectionData = {
+    country: string;
+    edition: string;
+    homeCurrency: string;
+    isMultiCurrencyEnabled: boolean;
+
+    journalEntryAccounts: Account[];
+    bankAccounts: Account[];
+    creditCards: Account[];
+    accountsReceivable: Account[];
+    accountsPayable: Account[];
+    otherCurrentAssetAccounts: Account[];
+
+    taxCodes: TaxCode[];
+    employees: Employee[];
+    vendors: Vendor[];
+};
+
+type IntegrationEntityMap = 'NONE' | 'DEFAULT' | 'TAG' | 'REPORT_FIELD';
+
+type QBOConnectionConfig = {
```

Comment:
Since this is using an acronym, let's add a comment to describe what it means?

```suggestion
/**
 * User configuration for the QuickBooks Online accounting integration.
 */
type QBOConnectionConfig = {
```